File: src/tests/fixtures/QA-CS-105/must-not-fire/ProseMentions.cs
using Microsoft.Playwright.NUnit;
using NUnit.Framework;

public class ModalTests : PageTest
{
    [Test]
    public async Task ShouldOpenModal()
    {
        // A comment mentioning Page.WaitForTimeoutAsync(2000) must NOT fire.
        string doc = "see Page.WaitForTimeoutAsync(2000) usage";
        await Page.ClickAsync("button#open");
        await Expect(Page.GetByText("Modal")).ToBeVisibleAsync();
    }

    [Test]
    public void WaitForTimeoutAsyncIsOnlyMentionedInProse()
    {
        // The helper METHOD below is the FIX (an auto-waiting waitFor),
        // not a sleep — the lexical path could not tell declarations from
        // calls; the grammar path sees no invocation_expression.
        Assert.That(typeof(PageTest).Name, Is.Not.Empty);
    }
}

File: src/tests/fixtures/QA-JV-105/must-not-fire/ProseMentions.java
import org.junit.jupiter.api.Test;

class ModalTest {

    @Test
    void shouldOpenModal() {
        // A comment mentioning page.waitForTimeout(2000) must NOT fire.
        String doc = "see page.waitForTimeout(2000) usage";
        page.click("button#open");
        assertThat(page.getByText("Modal")).isVisible();
    }

    @Test
    void hasAHelperNamedWaitForTimeoutButNeverCallsIt() {
        // A method DECLARED with the name in prose-like position — the
        // lexical path would match `.waitForTimeout(` only when invoked,
        // so this pins the grammar path: no call node, no finding.
        assertThat(documentation()).isTrue();
    }

    private boolean documentation() {
        return true;
    }
}
